roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -425,15 +427,15 @@ def resolve_address(self, addr):
         """Look up the given IPv4 address using getent."""
         command = ' '.join(['/bin/getent', 'ahostsv4', addr])
         try:
-            out = self.read_command_output(command)
+            out = self.read_command_output(command, False)
```

Comment:
We are passing utf8=False here because we have a test case for resolve_address() that runs on incompatible python version?
This seems fragile instead why not detect if the given string result['stdout'] is encodable and act accordingly?
Something like:
```
try:
  output = result['stdout'].encode('utf-8')
except UnicodeEncodeError:
  output = result['stdout']
return output
```

Reply:
`read_command_output` encodes the output as UTF-8 to avoid problems saving log files containing non-ASCII characters.  In Python 2, this encoding presents no problems; UTF-8-encoded strings look like regular strings and behave as expected in comparisons:
```
>>> 'foo'.encode('utf-8')
'foo'
>>> 'foo'.encode('utf-8') == 'foo'
True
```
In Python 3, encoding the string into UTF-8 produces a "byte string", which looks and behaves differently:
```
>>> 'foo'.encode('utf-8')
b'foo'
>>> 'foo'.encode('utf-8') == 'foo'
False
```
When I tested with Python 2, everything was fine.  However, CI tests with both Python 2 and Python 3, and in the latter case, the test reveals that `resolve_address` is comparing incompatible string types.

An alternative to adding the `utf8` parameter would be to use byte strings in `resolve_address`.  Python 2 has forwards compatibility so that it accepts the byte-string syntax and treats a byte string as a regular string:
```
>>> b'foo' == 'foo'
True
```
So we could use `record[1] == b'STREAM'`, and it would make no different on Python 2 (where the code already works properly) and would fix the problem on Python 3 (where encoding the output as UTF-8 turns it into a byte string).  However, I decided to add the `utf8` parameter instead because it lets us avoid the weird byte-string syntax and the unnecessary encoding.